interp/interp_test.go

```diff
@@ -3427,19 +3426,55 @@ var testBuiltinsMap = map[string]func(HandlerContext, []string) error{
 		return os.Link(oldname, newname)
 	},
 	"touch": func(hc HandlerContext, args []string) error {
+		filenames := args // create all arugments as filenames
+
 		newTime := time.Now()
-		if args[0] == "-d" {
-			if !strings.HasPrefix(args[1], "@") {
-				return fmt.Errorf("unimplemented")
+		if args[0] == "-t" {
+			if len(args) < 3 {
+				return fmt.Errorf("usage: touch [-t [[CC]YY]MMDDhhmm[.SS]] file")
+			}
+			filenames = args[2:] // treat the rest of the args as filenames
+
+			arg := args[1]
+			if strings.HasPrefix(arg, "-") {
+				return fmt.Errorf("usage: touch [-t [[CC]YY]MMDDhhmm[.SS]] file")
+			}
+			var offset int // index after yy
+			var (          // optional date fields
+				yy    = "2022"
+				ss    = "00"
+				chunk = 2 // per-date field chunk size
+			)
+			if strings.Contains(arg, ".") {
+				ss = arg[len(arg)-chunk:]
+			}
+			if len(arg) < 12 {
+				// assume MMDDhhmm or MMDDhhmm.SS
+			} else if len(arg) >= 12 {
+				// assume CCYYMMDDhhmm or CCYYMMDDhhmm.SS
+				yy = arg[:4]
+				offset = len(yy)
+			} else if len(arg) > 15 {
+				return fmt.Errorf("usage: touch [-t [[CC]YY]MMDDhhmm[.SS]] file")
 			}
-			sec, err := strconv.ParseInt(args[1][1:], 10, 64)
+
+			var ( // field start indices
+				month = offset
+				dd    = month + chunk
+				hh    = dd + chunk
+				mm    = hh + chunk
+			)
+			v := func(i int) string {
+				return arg[i : i+chunk]
+			}
+			s, err := time.Parse(time.RFC3339, fmt.Sprintf("%s-%s-%sT%s:%s:%sZ", yy, v(month), v(dd), v(hh), v(mm), ss))
 			if err != nil {
 				return err
 			}
-			newTime = time.Unix(sec, 0)
-			args = args[2:]
+			newTime = s
 		}
-		for _, arg := range args {
+
+		for _, arg := range filenames {
```

Comment:
nit: also error if any of the filenames begin with a dash

Reply:
nice catch, bash would actually error. added check.